src/ExposureHistory/History/DateInfoHeader.tsx

```diff
@@ -29,13 +41,75 @@ const DateInfoHeader: FunctionComponent<DateInfoHeaderProps> = ({
     ? determineUpdatedAtText(lastDetectionDate)
     : ""
 
+  const handleRefreshButtonPressed = async () => {
+    setRefreshing(true)
+    await checkForNewExposures()
+    setRefreshing(false)
+  }
+
+  const rotation = new Animated.Value(0)
+
+  const animatedStyle = {
+    rotatingIcon: {
+      transform: [
+        {
+          rotate: rotation.interpolate({
+            inputRange: [0, 1],
+            outputRange: ["0deg", "360deg"],
+          }),
+        },
+      ],
+    },
+  }
+
+  Animated.loop(
```

Comment:
@johnschoeman let me know what you think of this, totally ok removing it if you think adds unnecessary complication 🧐

Reply:
Its a neat feature, I would say its a nice to have, but you've already coded it, so we should ship it!

and while it does add complexity, it is self contained so not the type of complexity I would be worried about leaking around the system.